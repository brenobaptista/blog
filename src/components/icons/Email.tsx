interface Props {
  width: number
  height: number
}

const Email = ({ width, height }: Props) => (
  <svg
    xmlns='http://www.w3.org/2000/svg'
    viewBox='0 0 24 24'
    width={width}
    height={height}
    fill='none'
    stroke='currentColor'
    strokeWidth='2'
    strokeLinecap='round'
    strokeLinejoin='round'
    aria-hidden='true'
  >
    <title>Contact me by email</title>
    <path d='M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z' />
    <polyline points='22,6 12,13 2,6' />
  </svg>
)

export default Email
