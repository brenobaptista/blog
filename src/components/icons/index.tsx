export { default as Email } from './Email'
export { default as GitHub } from './GitHub'
export { default as LinkedIn } from './LinkedIn'
export { default as Magnifier } from './Magnifier'
export { default as Portrait } from './Portrait'
export { default as RSS } from './RSS'
