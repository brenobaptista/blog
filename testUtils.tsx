import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
