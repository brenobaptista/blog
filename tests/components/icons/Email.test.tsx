import { render } from 'testUtils'
import { Email } from '@/icons'

test('matches snapshot', () => {
  const { asFragment } = render(<Email width={40} height={40} />)

  expect(asFragment()).toMatchSnapshot()
})
