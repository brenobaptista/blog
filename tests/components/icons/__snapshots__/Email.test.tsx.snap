// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    fill="none"
    height="40"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Contact me by email
    </title>
    <path
      d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
    />
    <polyline
      points="22,6 12,13 2,6"
    />
  </svg>
</DocumentFragment>
`;
